### ISSUE TITLE
Dotted circle between HALANT and CONS_FINAL_MOD

### ISSUE THREAD

**Issue by dscorbett** [behdad/harfbuzz#478]
When a CONS_FINAL_MOD follows a HALANT, HarfBuzz inserts a dotted circle between them; for example, ⟨த்³⟩. This is in accord with the USE specification, but the specification is wrong, at least for the superscript and subscript digits used in Tamil.

**Comment by behdad** [comment]
Can you references Unicode text on this?

**Comment by dscorbett** [comment]
What little Unicode says about this is [in chapter 12](http://www.unicode.org/versions/Unicode9.0.0/ch12.pdf#page=42):

> The Tamil script has fewer consonants than the other Indic scripts. When representing the “missing” consonants in transcriptions of languages such as Sanskrit or Saurashtra, superscript European digits are often used, so ப² = _pha_, ப³ = _ba_, and ப⁴ = _bha_. The characters U+00B2, U+00B3, and U+2074 can be used to preserve this distinction in plain text.

It does not describe how digits interact with virama or vowel signs, but such combinations [are attested](http://www.unicode.org/L2/L2010/10436-infitt-resp.pdf#page=12).

**Comment by behdad** [comment]
Also note that Tamil does NOT go through USE shaper.

**Comment by behdad** [comment]
Here's Andrew Glass's comment re this issue (via private mail):
```
For the latest issue, the bug report is correct (aside from the fact that Tamil doesn't get handled by USE in either Harfbuzz or Uniscribe). That is to say, items having ISC =  Syllable_modifier should be allowed to combine with Virama, since a virama terminated cluster is still a viable orthographic syllable (whether or not it is actually pronounceable). So the change to make here is to modify the virama terminated cluster definition to allow FM after H:


[< R | CS >] < B | GB > [VS] (CMAbv)* (CMBlw)* (< H B | SUB > [VS] (CMAbv)* (CMBlw)*)* H [FM]


That makes me wonder if we should allow multiple FM. I've not heard of the need, but in the abstract this could occur at least with the Tibetan syllable modifiers. So if I were going to update the spec and the code, I would add that flexibility to both the standard and virama-terminated clusters  . . . (FM)*
```

**Comment by dscorbett** [comment]
One reason to allow multiple FMs is that U+1A7F TAI THAM COMBINING CRYPTOGRAMMIC DOT may appear many times on one base.

**Comment by Richard57** [comment]
Except that U+1A7F is actually a Nukta - to which it might be corrected this week.

<RA HAAM, MAI SAM> ought to be another example - but (a) I can't find any instances and (b) their categories may also be corrected this week (UTC #151).
